app/utils/file/picker_util.ts

```diff
@@ -0,0 +1,360 @@
+
+        if (hasCameraPermission) {
+            launchCamera(options, async (response: FileResponse) => {
+                StatusBar.setHidden(false);
+
+                if (response.error || response.didCancel) {
+                    return;
+                }
+
+                const files = await this.getFilesFromResponse(response);
+                await this.prepareFileUpload(files);
+            });
+        }
+    };
+
+    attachFileFromFiles = async (browseFileType?: string) => {
+        const hasPermission = await this.hasStoragePermission();
+        const fileType = browseFileType ?? Platform.select({ios: 'public.item', android: '*/*'})!;
+
+        if (hasPermission) {
+            try {
+                const res = (await DocumentPicker.pickSingle({type: [fileType]})) as unknown as File;
+
+                if (Platform.OS === 'android') {
+                    // For android we need to retrieve the realPath in case the file being imported is from the cloud
+                    const newUri = await ShareExtension.getFilePath(res.uri);
+                    if (newUri.filePath) {
+                        res.uri = newUri.filePath;
+                    } else {
+                        return;
+                    }
+                }
+
+                // Decode file uri to get the actual path
+                res.uri = decodeURIComponent(res.uri);
+
+                await this.prepareFileUpload([res]);
+            } catch (error) {
+                // Do nothing
+            }
+        }
+    };
+
+    attachFileFromPhotoGallery = async () => {
+        const options: ImageLibraryOptions = {
+            quality: 1,
+            mediaType: 'mixed',
+            includeBase64: false,
+        };
+
+        if (Platform.OS === 'ios') {
+            options.mediaType = 'mixed';
+        }
+
+        const hasPermission = await this.hasPhotoPermission('photo');
+
+        if (hasPermission) {
+            launchImageLibrary(options, async (response: FileResponse) => {
+                StatusBar.setHidden(false);
+                if (response.error || response.didCancel) {
+                    return;
+                }
+
+                const files = await this.getFilesFromResponse(response);
+                await this.prepareFileUpload(files);
+            });
+        }
+    };
+
+    hasPictureUrl = (userId: string, lastPictureUpdate: number, serverUrl: string) => {
+        let client: Client | undefined;
+        let profileImageUrl: string | undefined;
+
+        try {
+            client = NetworkManager.getClient(serverUrl);
+            profileImageUrl = client.getProfilePictureUrl(userId, lastPictureUpdate);
+        } catch {
+            return false;
+        }
+
+        // Check if image url includes query string for timestamp. If so, it means the image has been updated from the default, i.e. '.../image?_=1544159746868'
+        return Boolean(profileImageUrl?.includes('image?_'));
+    };
```

Comment:
huh? what is going on here? `client/getProfilePictureUrl` is a local function that constructs the url, this will ALWAYS be true!!!